modules/Router.js

```diff
@@ -15,5 +15,48 @@
    * @class  Router
    * @author kent.willis@vecna.com
    */
-  return Backbone.Router.extend({});
+
+  var Router = Backbone.Router.extend({
+
+    /**
+    * overridden the route function to send start and end times to Logger
+    */
+    route: function(route, name, callback) {
+      if (!_.isRegExp(route)) route = this._routeToRegExp(route);
+      if (_.isFunction(name)) {
+        callback = name;
+        name = '';
+      }
+      if (!callback) callback = this[name];
+      var router = this;
+      Backbone.history.route(route, function(fragment) {
+
+        newrelic.setCustomAttribute('route', name);
+        var UUID = "uuid-"+(new Date()).getTime().toString(16)+Math.floor(1E7*Math.random()).toString(16);
+        Logger.track({
+          UUID : UUID,
+          type : "routeChange",
+          state: "start",
+          route: name,
+          time: Date.now(),
+        });
+
+        var args = router._extractParameters(route, fragment);
```

Comment:
I'm not sure what you mean by 'chainable, generic hooks'. Can you give me an example?